ui/ozone/platform/wayland/wayland_window.cc

```diff
@@ -96,10 +123,33 @@ bool WaylandWindow::Initialize() {
   return true;
 }
 
+void WaylandWindow::CreateXdgPopup() {
+  if (!parent_window_)
+    parent_window_ = connection_->GetCurrentFocusedWindow();
```

Comment:
Check comment at https://github.com/Igalia/chromium/pull/428/commits/1807c9636035bc23c34d5a3176199cf68ec4b44b#diff-ab710017f2cbbeef686bc6cc4c5a338cR60

Reply:
Looks like you didn't pay attention to this.

Would you please check the following commit - 4de6b42a0ee104503cfb2763a006784f1812b836 ?

Afaik, Widget::InitParams has a  gfx::NativeView parent, which you can reuse to extract a gfx::AcceleratedWidget from it. Then, you could call delegate_->GetParentWindow(gfx::AcceleratedWidget* parent_widget) and take it instead. Once you have it, just call WaylandConnection::GetWindow(widget) to get a parent WaylandWindow.

Reply:
Looks like you didn't pay attention to this.

Would you please check the following commit - 4de6b42 ?

Afaik, Widget::InitParams has a gfx::NativeView parent, which you can reuse to extract a gfx::AcceleratedWidget from it. Then, you could call delegate_->GetParentWindow(gfx::AcceleratedWidget* parent_widget) and take it instead. Once you have it, just call WaylandConnection::GetWindow(widget) to get a parent WaylandWindow.

Reply:
@msisov, please look into Issue #424. I created issue for parent-child relationship based on pads we listed up tasks and it's prepared at my local in order to send another PR once this PR is merged. I thought we'd better keep previous history. Do you think we have to handle it with this branch together?

Reply:
I'd better want you to do it right way here rather than duplicating our efforts. It will ease upstreaming efforts. I'm 100% sure that the way how current patch does won't be accepted.